### DISCUSSION TITLE
Inconsistent RF data - Sonoff RF Bridge R2(v2.2)

### DISCUSSION THREAD

**Discussion by db-rbt** [arendst/Tasmota#14783] [Q&A] [answered]
Hello,
I am having difficulties with a new Sonoff RF Bridge R2 (v2.2). Flashed Tasmota (but not Portsich) -- the RF data received by Tasmota is not consistent -- it changes.

Tried 2 devices, a Sonoff PIR3 and a Sonoff RF Remote, and in both instances the RF data reported was inconsistent, changing with almost every reception. Here are the logs:

PIR3:

> {"Sync":14750,"Low":578,"High":1666,"Data":"**CD2540**","RfKey":"None"}
> {"Sync":12810,"Low":580,"High":1664,"Data":"**CD2540**","RfKey":"None"}
> {"Sync":14738,"Low":608,"High":1634,"Data":"**CD0540**","RfKey":"None"}
> {"Sync":14782,"Low":620,"High":1622,"Data":"**4D0540**","RfKey":"None"}
> {"Sync":14760,"Low":604,"High":1624,"Data":"**0D2140**","RfKey":"None"}
> {"Sync":12806,"Low":576,"High":1666,"Data":"**CD2540**","RfKey":"None"}
> {"Sync":14734,"Low":584,"High":1660,"Data":"**C52500**","RfKey":"None"}
> {"Sync":12856,"Low":580,"High":1664,"Data":"**CC2540**","RfKey":"None"}
> {"Sync":12846,"Low":580,"High":1662,"Data":"**CD2540**","RfKey":"None"}
> {"Sync":14744,"Low":604,"High":1640,"Data":"**452140**","RfKey":"None"}
> {"Sync":14750,"Low":620,"High":1622,"Data":"**4C2440**","RfKey":"None"}
> {"Sync":14748,"Low":596,"High":1648,"Data":"**8D2540**","RfKey":"None"}
> {"Sync":12830,"Low":582,"High":1662,"Data":"**CD2540**","RfKey":"None"}
> {"Sync":13402,"Low":586,"High":1658,"Data":"**C52540**","RfKey":"None"}
> 

RF Remote:

> {"Sync":23116,"Low":716,"High":2226,"Data":"**DC8026**","RfKey":"None"}
> {"Sync":21930,"Low":730,"High":2216,"Data":"**DC8022**","RfKey":"None"}
> {"Sync":21370,"Low":726,"High":2220,"Data":"**DC8026**","RfKey":"None"}
> {"Sync":23120,"Low":724,"High":2220,"Data":"**DC8026**","RfKey":"None"}
> {"Sync":21366,"Low":754,"High":2194,"Data":"**DC8026**","RfKey":"None"}
> {"Sync":21374,"Low":726,"High":2222,"Data":"**DC8026**","RfKey":"None"}
> {"Sync":21920,"Low":726,"High":2222,"Data":"**DC8026**","RfKey":"None"}
> {"Sync":23564,"Low":730,"High":2216,"Data":"**DC8026**","RfKey":"None"}
> {"Sync":23524,"Low":726,"High":2218,"Data":"**DC8026**","RfKey":"None"}
> {"Sync":23428,"Low":724,"High":2218,"Data":"**DC8026**","RfKey":"None"}
> {"Sync":23580,"Low":726,"High":2220,"Data":"**DC8026**","RfKey":"None"}
> {"Sync":23136,"Low":766,"High":2186,"Data":"**9CA006**","RfKey":"None"}
> {"Sync":21456,"Low":756,"High":2196,"Data":"**DC8006**","RfKey":"None"}
> 

The PIR is very simple and sends a single code (there is no OFF code). The Remote also sends a single code when activated. So, in both instances the Bridge should consistently receive a single code.

Interestingly in both cases, the RF codes, while not constant, had a pattern (PIR usually contains  a “45” and a “C’ or “D” or both).

These types of patterns are usually indicative of a buffer overrun, an inconsistently initialized variable, multiple variables clobbering  the same memory location, or some other inadvertent side effect. 

Has anyone else encountered this problem?

Thanks!

**Comment by sfromis** [comment]
You "forgot" the most typical cause of data varying but still having a pattern, receiver not successfully getting/parsing the data the transmitter tried to send. Risk of this increases in case of the transmitter being a very simple device, maybe not very exact about specs. Or RF noise sources active nearby. Or....

**Reply by db-rbt** [reply]
Thanks for the suggestion. 
I did not forget the miscommunication problem. I did try in in several locations, as well as with 2 versions of Tasmosta -- no change. If it is a miscommunication it would probably  mean  a hardware problem, but I wanted to see if anyone in the community encountered similar symptoms, and their causes. It could also be induced by the hardware  changes in v2.2, perhaps using a different data format or encoding and Tasmota not having the correct algorithm for it.

**Comment by Jason2866** [comment, marked as answer]
A lot of assumptions and suppositions but nothing concrete to get to the bottom of the possible cause.
I just can say my two Sonoff RfBridges (R1 and R2) do work. Yes, nothing concrete too, but enough to say it is working in general.
Both are flashed with Portisch firmware

**Reply by db-rbt** [reply]
Thanks for your feedback. Do you have the R2 or what they call R2 v2.2?
I noticed that in the v2.2 they replaced the antenna coils with an antenna trace on the PCB -- I would guess that the PCB antenna is not as good as the coil, so that could be a reason.
I also noticed some improvement with the latest Tasmota dev firmware.. will post more a bit later -- so that could also be a factor,

**Comment by db-rbt** [comment]
I think I narrowed down the source of the problem, and it seems to be Tasmota:
Since I posted this, I flashed the latest dev firmware 10.1.08, and the RFdata decodes improved significantly. 

One of the changes incorporated into this version is a fix for the serial logic in the underlying Arduino code:

[https://github.com/arendst/Tasmota/discussions/14783.](url)

This fix was for a problem with a Zigbee device, but it seems to have partially  helped with my problem. The Rf Data decode is still not perfect but the hit rate rose from 20-30% to about 60%. It is possible that there is still some lingering issue in the code.
